tests/field_defaults/tests.py

```diff
@@ -169,6 +170,17 @@ def test_bulk_create_mixed_db_defaults_function(self):
         years = DBDefaultsFunction.objects.values_list("year", flat=True)
         self.assertCountEqual(years, [2000, datetime.now().year])
 
+    def test_full_clean(self):
+        obj = DBDefaults()
```

Comment:
I changed it to `DBArticle` because `DBDefaults.both` also has `default` defined.

Reply:
Tests with `DBArticle` proves that we should backport this change as `Model.full_clean()` can crash on expressions:
```
  File "/django/tests/field_defaults/tests.py", line 175, in test_full_clean
    obj.full_clean()
  File "/django/django/db/models/base.py", line 1586, in full_clean
    self.clean_fields(exclude=exclude)
  File "/django/django/db/models/base.py", line 1641, in clean_fields
    setattr(self, f.attname, f.clean(raw_value, self))
  File "/django/django/db/models/fields/__init__.py", line 830, in clean
    value = self.to_python(value)
  File "/django/django/db/models/fields/__init__.py", line 1620, in to_python
    parsed = parse_datetime(value)
  File "/django/django/utils/dateparse.py", line 114, in parse_datetime
    return datetime.datetime.fromisoformat(value)
TypeError: fromisoformat: argument must be str
```